src/__tests__/test-utils-selectors.test.tsx

```diff
@@ -0,0 +1,141 @@
+            if (path.node.property.type === 'Identifier') {
+              parseNested(path.node.object, path.node.property.name);
+            } else if (path.node.property.type === 'StringLiteral') {
+              parseNested(path.node.object, path.node.property.value);
+            } else if (path.node.property.type === 'TemplateLiteral') {
+              parseNested(path.node.object, buildTemplateString(path.node.property));
+            } else {
+              throw new Error('Unhandled selectors access type.');
+            }
+          }
+        },
+      },
+    } as PluginObj;
+  }
+  const source = fs.readFileSync(file, 'utf-8');
+  transformSync(source, { babelrc: false, configFile: false, plugins: [extractor] })?.code;
+}
+
+function extractComponentSelectors(file: string, usedProperties: string[], onExtract: (selector: string) => void) {
+  function extractor(): PluginObj {
+    return {
+      visitor: {
+        ObjectProperty(path: NodePath<types.ObjectProperty>) {
+          if (path.node.key.type === 'StringLiteral' && matchProperties(usedProperties, path.node.key.value)) {
+            if (path.node.value.type === 'StringLiteral') {
+              onExtract(trimSelectorHash(path.node.value.value));
+            } else {
+              throw new Error('Unexpected selector value format');
+            }
+          }
+        },
+      },
+    } as PluginObj;
+  }
+  const source = fs.readFileSync(file, 'utf-8');
+  transformSync(source, { babelrc: false, configFile: false, plugins: [extractor] })?.code;
+}
+
+function trimSelectorHash(selector: string) {
+  const splitSelector = selector.replace('.', '').split('_');
+  if (splitSelector.length >= 5) {
+    splitSelector.splice(splitSelector.length - 2, splitSelector.length);
+    return splitSelector.join('_');
+  }
+  return selector;
+}
+
+function getComponentNameFromFilePath(filePath: string) {
+  return filePath.match(/lib\/components\/([\w-]+)/)![1];
+}
+
+// Build string literal from template string replacing arguments with wildcards ("*").
+function buildTemplateString(node: types.TemplateLiteral) {
+  let literal = '';
+  for (const element of flatten(zip(node.quasis, node.expressions))) {
+    if (!element) {
+      continue;
+    } else if (element.type === 'TemplateElement') {
+      literal += element.value.raw;
+    } else if (element.type === 'Identifier') {
+      literal += '*';
+    } else if (element.type === 'NumericLiteral') {
+      literal += element.value;
+    } else {
+      throw new Error('Unhandled template literal structure.');
+    }
+  }
+  return literal;
+}
+
+// Match property against the used properties supporting the wildcards ("*").
+function matchProperties(usedProperties: string[], property: string) {
+  for (const testProperty of usedProperties) {
+    if (testProperty === property) {
+      return true;
+    }
+    if (testProperty.includes('*') && new RegExp(testProperty.replace(/\*/g, '.*')).test(property)) {
```

Comment:
Dynamically constructing regexp is unsafe, because there could be other special characters which can be treated as control statements. 

Learn how to write safe regexps and update

Reply:
How much do we need it for this case? Yes, there can be in theory some weird template string so that the resulting expression will match more or less properties in the worst scenario.

Reply:
If you want to have a weaknesses in your solution, I cannot do anything with that

Reply:
But at least you could add `^` and `$` to ensure it matches the full strings only